script/check-broken-links.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-param-reassign */
+const createBrokenLinkChecker = require('metalsmith-broken-link-checker');
+
+const pagesOutsideOfContent = [
+  '/employment/job-seekers/skills-translator',
+  '/employment/job-seekers/create-resume',
+  '/employment/employers'
+];
+
+function checkBrokenLinks() {
+  return (files, metalsmith, done) => {
+    const ignorePaths = [
+      ...pagesOutsideOfContent
+    ];
+
+    for (const fileName of Object.keys(files)) {
+      const file = files[fileName];
+
+      const {
+        entryname,
+        path
+      } = file;
+
+      const isApp = !!entryname;
+      if (isApp) ignorePaths.push(path);
+    }
+
+    const ignoreGlobs = ignorePaths.map(path => `${path}(.*)`);
+    const ignoreLinks = new RegExp(ignoreGlobs.join('|'));
+    const brokenLinkChecker = createBrokenLinkChecker({
+      allowRedirects: true,
+      warn: false,
+      allowRegex: ignoreLinks
+    });
+
+    brokenLinkChecker(files);
+    done();
+  };
```

Comment:
Previously we had a hardcoded array of paths to not evaluate as link-values during builds. We used this for React routers, which sometimes has child paths that otherwise don't exist in the file system so would otherwise cause an error. This was confusing with the `brand-consolidation` branch, which is renaming React applications. To account for this, this middleware generates the list by gathering all file paths for file's that contain an entry name for Webpack bundles.